app/models/teacher_set.rb

```diff
@@ -901,6 +914,23 @@ def parse_items_available_and_total_count(response)
     return total_count, available_count
   end
 
+
+  # Call sierra bib api to get set_type value
+  def get_set_type_value_from_bib_response(bibid)
+    resp = send_request_to_bibs_microservice(bibid)
+    set_type = nil
+    if resp.present? && resp.code == 200
+      resp["data"].each do |bib_record|
+        set_type_marctag = bib_record['varFields'].detect { |hash| hash['marcTag'] == '526' }
+        if set_type_marctag.present?
+          set_type = set_type_marctag['subfields'].map { |x| x['content']}.join(', ')
+        end
+      end
+    end
+    set_type
```

Comment:
Is there a default set_type?  If you don't get a meaningful response, what would you like to set the type to?  Or would you prefer to keep the column un-filled, and do the default value calculation later, at the elastic search stage?

Reply:
@darya  Calling correct endpoint and checking with response code (resp.present? && resp.code == 200)  .Requirement is what ever value is there in 526 we need to retrieve and save.
If 526 field itself not present can you check update_set_type method in teacher_set.rb. Already this scenarios worked in last sprint.